### DISCUSSION TITLE
Play 2.9.0-M6, Scala 3.3.0, SBT 1.9.2 library conflicts

### DISCUSSION THREAD

**Discussion by polydevuk** [playframework/playframework#11912] [Q&A]
Just used g8 to setup a Play project using Play 2.9.0-M6, Scala 3.3.0 & SBT 1.9.2 but I'm getting library conflict errors:

```
[error] java.lang.RuntimeException: found version conflict(s) in library dependencies; 
[error] some are suspected to be binary incompatible:
[error]
[error] 	* org.scala-lang.modules:scala-xml_2.12:2.1.0 (early-semver) is selected over 1.3.0
[error] 	    +- com.typesafe.play:twirl-api_2.12:1.6.0-RC4         (depends on 2.1.0)
[error] 	    +- com.github.sbt:sbt-native-packager:1.9.16 (scalaVersion=2.12, sbtVersion=1.0) (depends on 2.1.0)
[error] 	    +- org.foundweekends.giter8:giter8-lib_2.12:0.13.1    (depends on 1.3.0)
```

I added this to `build.sbt` but to no avail:

```
libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
libraryDependencySchemes += "com.typesafe.play" %% "twirl-api" % VersionScheme.Always
libraryDependencySchemes += "com.github.sbt" %% "sbt-native-packager" % VersionScheme.Always
libraryDependencySchemes += "org.foundweekends.giter8" %% "giter8-lib" % VersionScheme.Always
```

Ok, I finally got it working. Here's how:

- Add this to `project/plugins.sbt`
  ```
  ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
  ```

- Change the Scala Test dependency version:
  ```
  // libraryDependencies += "org.scalatestplus.play" %% "scalatestplus-play" % "5.1.0" % Test
  libraryDependencies += "org.scalatestplus.play" %% "scalatestplus-play" % "6.0.0-M6" % Test
  ```

- In IntelliJ Edit Configurations select JDK 17 as this overrides your env `JAVA_HOME`

- Reduce JVM options in IntelliJ project config to:
  ```
  -Xms512M -Xmx1024M -Xss1M
  ```

**Comment by mkurz** [comment]
When using `sbt new` it currently generated a Play 2.8 project, so yes you need to updated some dependencies to make things work.
However you could just run:
```
sbt new playframework/play-scala-seed.g8 --branch 2.9.x
```
Then it uses the 2.9.x branch of the scala-seed project which sets things up correctly for Play 2.9

**Reply by mkurz** [reply]
In your case you should upgrade the `sbt-giter8-scaffold` to  0.16.2 in your `project/plugins.sbt`, then you would not need `libraryDependencySchemes` configs, since that version comes with latest scala-xml.
But like said, just use the `--branch 2.9` switch to generate the correct project.

**Reply by nemoo** [reply]
To try play  2.9, I also created a play 2.9 template via:
`sbt new playframework/play-scala-seed.g8 --branch 2.9.x`

In the generated files, in project/plugins.sbt, I now find something about giter8:
`addSbtPlugin("org.foundweekends.giter8" % "sbt-giter8-scaffold" % "0.16.2")`

I am confused. Shouldn't the generated play app just be a play app without dependencies on giter8?

**Reply by mkurz** [reply]
The filters plugin is required for scaffolding, see https://github.com/playframework/play-scala-seed.g8#scaffolding

btw, `sbt new playframework/play-scala-seed.g8` is enough now to create a 2.9 project. If using SBT 1.9, `sbt new` does also work.
